funcframework/framework.go

```diff
@@ -109,6 +109,21 @@ func Start(port string) error {
 	return http.ListenAndServe(":"+port, server)
 }
 
+// StartContext serves an HTTP server with registered function(s), shutting the server
+// down when the context is cancelled.
+func StartContext(ctx context.Context, port string) error {
+	server, err := initServer()
+	if err != nil {
+		return err
+	}
+	srv := &http.Server{Addr: ":" + port, Handler: server}
+	go func() {
+		<-ctx.Done()
+		srv.Shutdown(context.Background())
+	}()
+	return http.ListenAndServe(":"+port, server)
+}
+
 func initServer() (*http.ServeMux, error) {
```

Comment:
Hello, thanks for your contribution!

It seems like exposing `initServer` and changing the name to `NewServer` would be a more flexible solution. That would make it compatible with the `httptest` package too and avoid the redundancy of two Start functions.